docs/tutorial/miscellany.rst

```diff
@@ -276,7 +276,7 @@ type ``Int`` will be provided. Otherwise, it will provide the type
     strToType _ = Nat
 
     fromFile : String -> IO (Provider Type)
-    fromFile fname = do str <- readFile fname
+    fromFile fname = do Right str <- readFile fname
```

Comment:
I think it might be a good idea to handle the error like in https://github.com/idris-lang/Idris-dev/blob/b3fdc3e2dc6256b9cb03e4cee89edb1dd97b081d/test/ffi004/test026.idr#L14 .